src/firefly/js/charts/ui/PlotlySaveDialog.jsx

```diff
@@ -41,47 +41,166 @@ export function showPlotLySaveDialog(Plotly, chartDiv) {
 }
 
 function generateTitleFromAxis(layout) {
-    const xaxis= layout.xaxis?.title?.text;
-    const yaxis= layout.yaxis?.title?.text;
-    return (isString(xaxis) || isString(yaxis)) ? `${isString(xaxis)?xaxis:''} ${isString(yaxis)?yaxis:''}` : undefined;
+    const xaxis = layout?.xaxis?.title?.text;
+    const yaxis = layout?.yaxis?.title?.text;
+    return (isString(xaxis) || isString(yaxis)) ? `${isString(xaxis) ? xaxis : ''} ${isString(yaxis) ? yaxis : ''}` : undefined;
+}
+
+function getDefaultFilename(chartDiv, fmt='png')  {
+    const layout = chartDiv?.layout;
+    const title = isString(layout?.title) ? layout.title : layout?.title?.text;
+    const axisTitle = generateTitleFromAxis(layout);
+    return makeDefaultDownloadFileName('chart', title ?? axisTitle, fmt);
 }
 
-function getDefaultFilename(chartDiv)  {
-    const {layout}= chartDiv;
-    const title=  isString(layout.title) ? layout.title : layout.title?.text;
-    const axisTitle= generateTitleFromAxis(layout);
-    return makeDefaultDownloadFileName('chart',title??axisTitle,'png');
+function ensureExt(name, ext) {
+    return name?.toLowerCase?.().endsWith(ext) ? name : `${name}${ext}`;
 }
 
+/**
+ * Get a plain Plotly figure: {data, layout, frames, config}
+ * @param Plotly
+ * @param chartDiv
+ */
+function getFigure(Plotly, chartDiv) {
+    if (chartDiv && (Array.isArray(chartDiv.data) || chartDiv.layout)) {
+        return {
+            data: Array.isArray(chartDiv.data) ? chartDiv.data : [],
+            layout: chartDiv.layout ?? {},
+            frames: chartDiv.frames ?? [],
+            config: chartDiv.config ?? {}
+        };
+    }
+    //else, nothing found
+    return { data: [], layout: {}, frames: [], config: {} };
```

Comment:
Same here.  The `if` condition only make it confusing.  I think you really just want:
```js
        return {
            data: chartDiv?.data ?? [],
            layout: chartDiv?.layout ?? {},
            frames: chartDiv?.frames ?? [],
            config: chartDiv?.config ?? {}
        };
```